plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDependency.java

```diff
@@ -234,6 +235,20 @@ public DBSObject getParentObject() {
      * SQL query originally copy-pasted from pgAdmin sources with some modifications.
      */
     public static List<PostgreDependency> readDependencies(DBRProgressMonitor monitor, PostgreObject object, boolean dependents) throws DBCException {
+        PostgreDataSource dataSource = object.getDataSource();
+        Boolean isMMode = false;
+        if (dataSource instanceof DatabaseCompatibilityProvider) {
+            DatabaseCompatibilityProvider compatibilityProvider = (DatabaseCompatibilityProvider) dataSource;
+            try {
+                String compatibilityMode = compatibilityProvider.getDatabaseCompatibleMode();
+                if ("M".equals(compatibilityMode)) {
+                    isMMode = true;
+                }
+            } catch (DBException e) {
+                log.error("Failed to get GaussDB compatibility mode", e);
+            }
+        }
```

Comment:
Maybe in future, GaussDB should not use postgresql ui or will couple GaussDB features with postgresql which postgresql do not have.

Reply:
I got a question about implementation. In future, GaussDB is not compatible with Postgres. If they share same code, modifications to Postgres will influence GaussDB, vice versa.  

Although GaussDB currently shares a significant amount of code with Postgres, implementing code separation might enhance stability, especially with community contributions.

Could the DBeaver team provide any suggestions on this?